### PR TITLE
#69 implement contains() for Directory

### PR DESCRIPTION
**Pull request by komape**
Implemented the Issue #69. Once as synchronous version, once as asynchronous. The code comments should be clear about usage. Nevertheless, I am available for questions.

### CODE REVIEW

File: test/directory.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import 'dart:io';
+
+import 'package:test/test.dart';
+import 'package:dartx/dartx_io.dart';
+
+void main() {
+  group('Directory', () {
+    test('contains', () async {
+      final thisDir = Directory.current;
```
Comment (passsy) [inline test/directory.dart:9]: Please use a temp directory when you create files during a test

File: test/directory.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      expect(await thisDir.contains(thisDir), false);
+      expect(await parentParentDir.contains(thisDir), false);
+      expect(await parentParentDir.contains(thisDir, recursive: true), true);
+
+      expect(await thisDir.contains(file), true);
+      expect(await thisDir.contains(file, recursive: true), true);
+      expect(await parentDir.contains(file), false);
+      expect(await parentDir.contains(file, recursive: true), true);
+
+      file.deleteSync();
```
Comment (passsy) [inline test/directory.dart:27]: wrap the deletion in `addTearDown` and place it right after creating the file.

Comment (komape) [reply test/directory.dart]: the temporary directory is now deleted recursively inside `addTearDown`

File: test/directory.dart (hunk 1 above)
Comment (komape) [reply test/directory.dart]: see new commit

**Comment by komape** [comment]
@passsy I don't understand why the test fail. I can't reproduce them on my local machine. Seems like the analyzer causes errors:

```log
2020-07-25T09:44:59.5932894Z Failed to precompile test:test:
2020-07-25T09:44:59.5933988Z /github/home/.pub-cache/hosted/pub.dartlang.org/analyzer-0.39.15/lib/src/dart/resolver/extension_member_resolver.dart:355:38: Error: The getter 'extensions' isn't defined for the class 'Scope'.
2020-07-25T09:44:59.5934476Z  - 'Scope' is from 'package:analyzer/dart/element/scope.dart' ('/github/home/.pub-cache/hosted/pub.dartlang.org/analyzer-0.39.15/lib/dart/element/scope.dart').
2020-07-25T09:44:59.5935491Z Try correcting the name to the name of an existing getter, or defining a getter or field named 'extensions'.
2020-07-25T09:44:59.5936059Z     for (var extension in _nameScope.extensions) {
2020-07-25T09:44:59.5936181Z                                      ^^^^^^^^^^
```

Do you know this error?

**Comment by komape** [comment]
@passsy can you rerun the tests? I can't do it. Only locally but there all tests pass:

```
user@device:~/code/dart/dartx$ dart test
Precompiling executable... (12.5s)
Precompiled test:test.
00:15 +310: All tests passed!
```

**Comment by leisim** [comment]
#115 solved the issue

**Pull request merged**